modules/js-and-css/optimize-oembeds/hooks.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Hook callbacks used for oEmbed Optimizer.
+ *
+ * @since n.e.x.t
+ * @package performance-lab
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$oembed_lazy_load_scripts = false;
+
+/**
+ * Filter the oEmbed HTML.
+ *
+ * Add loading="lazy" to any iframe tags.
+ * Lazy load any script tags.
+ *
+ * @since n.e.x.t
+ *
+ * @param string $html The oEmbed HTML.
+ * @return string
+ */
+function perflab_optimize_oembed_html( $html ) {
+	global $oembed_lazy_load_scripts;
+
+	$p = new WP_HTML_Tag_Processor( $html );
+
+	// Find the first iframe or script tag to act on.
+	while ( $p->next_tag() ) {
+		if ( 'IFRAME' === $p->get_tag() ) {
+			$loading_value = $p->get_attribute( 'loading' );
+			if ( empty( $loading_value ) ) {
+				$p->set_attribute( 'loading', 'lazy' );
+			}
+			return $p->get_updated_html();
+
+		} elseif ( 'SCRIPT' === $p->get_tag() ) {
+			$oembed_lazy_load_scripts = true;
+			$p->set_attribute( 'data-lazy-embed-src', $p->get_attribute( 'src' ) );
+			$p->set_attribute( 'src', '' );
+			return $p->get_updated_html();
+		}
+	}
```

Comment:
An embed may have both an `iframe` _and_ a `script`. For example, a Crowdsignal polls look like this:

```html
<script type="text/javascript" charset="utf-8" src="https://secure.polldaddy.com/p/7012505.js"></script>
<noscript><iframe title="Which design do you prefer?" src="https://poll.fm/7012505/embed" frameborder="0" class="cs-iframe-embed"></iframe></noscript>
```

So this wouldn't be handled properly here since the script would never get modified.

Reply:
Ah, good point and thanks for the example - I wasn't sure this was used. I'll test this without the `else` with a polldaddy embed.

Reply:
resolving as outdated, the new logic takes this into account.